mesonbuild/compilers/d.py

```diff
@@ -314,7 +314,9 @@ def _translate_args_to_nongnu(cls, args: T.List[str], info: MachineInfo, link_id
                 continue
             if arg.startswith('-fstack-protector'):
                 continue
-            if arg.startswith('-D'):
+            if arg.startswith('-D') and not (args == '-D' or args.startswith('-Dd') or args.startswith('-Df')):
```

Comment:
you can combine `str.startswith` checks
```suggestion
            if arg.startswith('-D') and not (args == '-D' or args.startswith(('-Dd', '-Df'))):
```

Is it only LDC that accepts these or does DMD accept them as well?

Reply:
yes, both LDC and DMD accept them.